### DISCUSSION TITLE
Custom installation directory for Chia

### DISCUSSION THREAD

**Discussion by doubleUTF** [Chia-Network/chia-blockchain#2671] [Ideas]
When chia is installed in Windows, by default the chosen installation path is the root drive, C. I notice when running a full node, the logs and especially the blockchain db can take up a sizeable space (2-3 Gb and counting). For various reasons, some may have reservations about letting Chia write logs and blockchain to the root drive (SSD wearing out, or low space). I could create a symlink to another suitable drive, but I think this feature should come out of the box considering the space is of variable length.

**Comment by mauricecyril** [comment]
On linux it looks like the hidden "~/.chia/mainnet/" directory location can be changed by editing the "default_root.py" file:
[PLEASE MAKE SURE YOU MAKE A BACKUP OF THE FILE AND ANY FOLDERS BEFORE ATTEMPTING THE STEPS BELOW. USE AT YOUR OWN RISK]
```
~/chia-blockchain/chia/util/default_root.py 
```

If you change "~/.chia/mainnet" to "[New Path]/.chia/mainnet" this will change the location that houses the DB (including the blockchain DB)

```
import os
from pathlib import Path

DEFAULT_ROOT_PATH = Path(os.path.expanduser(os.getenv("CHIA_ROOT", "~/.chia/mainnet"))).resolve()

```

Once this file has been updated run

```
cd ~/chia-blockchain/ && source venv/bin/activate
```

and

```
chia init
```

This is handy if you are using a Raspberry PI and want to move the DB folders to a different storage medium so that writes to an SD care are minimized.

**Reply by doubleUTF** [reply]
Thanks, this will be handy for Linux users; though it does require altering the python code.
My full node is on windows and I just created a symlink to point to another drive.
